src/Callback.php

```diff
@@ -4,110 +4,45 @@
 
 namespace Laminas\Filter;
 
-use Traversable;
+use Closure;
 
 use function array_unshift;
-use function call_user_func_array;
-use function class_exists;
 use function is_callable;
-use function is_string;
 
 /**
  * @psalm-type Options = array{
- *     callback?: callable,
- *     callback_params?: array,
- *     ...
+ *     callback: callable(mixed): mixed,
+ *     callback_params?: array<array-key, mixed>,
  * }
- * @extends AbstractFilter<Options>
+ * @implements FilterInterface<mixed>
  */
-final class Callback extends AbstractFilter
+final class Callback implements FilterInterface
 {
-    /** @var array */
-    protected $options = [
-        'callback'        => null,
-        'callback_params' => [],
-    ];
+    /** @var Closure(mixed): mixed */
+    private readonly Closure $callback;
+    private readonly array $arguments;
 
     /**
-     * @param callable|array|string|Traversable $callbackOrOptions
-     * @param array $callbackParams
+     * @param callable(mixed): mixed|Options $options
```

Comment:
Careful with this type definition: is this interpreted as `(callable(mixed): mixed)|Options`, or `callable(mixed): (mixed|Options)`?

Reply:
👍 Thanks - added parenthesis to clarify